### PR TITLE
Geoshp update

### PR DESCRIPTION
**Pull request by bingzhang**
need proxy setting in Clowder: custom.conf
clowder.proxy {
geoserver="http://admin:geoserver@yourhostip:8081/geoserver"
}

and geoserver and geoshp docker-compose.yml. Note it is depends on your local settings.     
PROXY_URL=http://yourhostip:9000/api/proxy/
PROXY_URL=http://yourhostip:9000/clowder/api/proxy/

version: "3.2"

services:
  geoserver:
    image: clowder/geoserver:${VERSION:-latest}
    networks:
      - clowder
    ports:
      - 8081:8080
    volumes:
      - /Users/bing/geoserver/generated/data/mounting:/data_dir
    restart: unless-stopped
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      restart_policy:
        condition: any 

  ncsa_geotiff_metadata:
      image: clowder/extractors-geoshp-preview:pyclowder2.3.4
      #image: geotiffmetadata
      networks:
        - clowder
      environment:
        - RABBITMQ_URI=${RABBITMQ_URI:-amqp://guest:guest@192.168.1.113/%2F}
        - RABBITMQ_EXCHANGE=${RABBITMQ_EXCHANGE:-clowder}
        - REGISTRATION_ENDPOINTS=http://yourhostip:9000/api/extractors?key=r1ek3rs
        - PROXY_ON=true
        - PROXY_HOST=http://yourhostip:9000/clowder/
        - PROXY_URL=http://yourhostip:9000/api/proxy/
        - GEOSERVER_USERNAME=admin
        - GEOSERVER_PASSWORD=geoserver
        - GEOSERVER_WORKSPACE=
        - GEOSERVER_USER=admin
        - GEOSERVER_URL=http://geoserver:8080/geoserver/
      depends_on:
        - geoserver
      restart: unless-stopped
      deploy:
        mode: replicated
        replicas: 1
        placement:
          constraints:
            - node.role == worker
        restart_policy:
          condition: any

networks:
 clowder:

### CODE REVIEW

**Comment by bingzhang** [comment]
the code can generate the wmslayer metadata, but the map is empty white. I think that is related to the box issue.

the bbox has some issue related to https://opensource.ncsa.illinois.edu/bitbucket/projects/CATS/repos/github-extractors-geo/browse/preview.geoshp/zipshputils.py#75.

Shall we use geoprocessing or gdal to get the box instead?

old code output
`WMS Layer URL: http://192.168.1.113:9000/clowder/api/proxy/geoserver/wms?request=GetMap&layers=5f9afe6882aea913e1aaf571:states.zip_5f9b1b1d82aea913e1aaf784&bbox=-13885038.383,2870337.13079,-7455049.48918,6338174.05576&width=640&height=480&srs=EPSG:3857&format=image%2Fpng`

new code output, 
`WMS Layer URL: http://192.168.1.113:9000/clowder/api/proxy/geoserver/wms?request=GetMap&layers=5f9afe6882aea913e1aaf571:states.zip_5f9b1b1d82aea913e1aaf784&bbox=inf,inf,5496036.399780443,-10147474.007654388&width=640&height=480&srs=EPSG:3857&format=image%2Fpng`

**Comment by robkooper** [comment]
looks like the whole bounding box is different, can you check and see if we are maybe missing the project libraries that are needed?

**Comment by max-zilla** [comment]
I'm still trying to understand, but `extent = self.findExtent()` already seems to use GDAL. It opens the file and reads the projection, bounding box, etc. using osgeo python library (aka GDAL):
```
        shpfile = ogr.Open(self.zipShpProp['shpFile'])
        osrs = osr.SpatialReference()
        osrs.ImportFromEPSG(int(self.zipShpProp['epsg']))
        dsrs = osr.SpatialReference()
        dsrs.ImportFromEPSG(3857) # google projeciton in epsg code
        ct = osr.CoordinateTransformation(osrs, dsrs)
        layer = shpfile.GetLayer(0)
        a = layer.GetExtent()
        proj = layer.GetSpatialRef()
        if proj.GetAttrValue("AUTHORITY", 1) == '4326':
            a = self.validateBbox(a)
        ab = ct.TransformPoint(a[0],a[2],0)
        cd = ct.TransformPoint(a[1],a[3],0)
        r= [ab[0], ab[1], cd[0], cd[1]]
```

in your example above, ab (the original file's extent) is returning (inf, inf) which seems like a bad input file. are those outputs both from the same input?

**Comment by bingzhang** [comment]
Screen Shot 2020-10-30 at 3 11 49 PM

File: preview.geoshp/Dockerfile [hunk 1]
```diff
@@ -9,14 +9,18 @@ ARG GITSHA1="unknown"
...
-    && pip install --upgrade  -r /home/clowder/requirements.txt \
-    && rm -rf /var/lib/apt/lists/*
+       python3 \
+       python-is-python3 \
+       python3-pip \
+       python3-gdal \
+       python3-rtree \
+       python3-scipy \
+       p7zip-full && \
+       pip3 install -r /home/clowder/requirements.txt && \
```
Comment (robkooper) [inline preview.geoshp/Dockerfile:22]: just as FYI, i like to do :

```
RUN install packages

COPY requirements.txt
RUN pip3 install -r requirements.txt
```

Advantage, if you just update requirements it does not break the cache until that line in the Dockerfile.

**Review by robkooper — APPROVED** [review]

**Pull request merged**